### PR TITLE
Allow transform to plain beans

### PR DESCRIPTION
**Pull request by erik-mythical**
Add method to transform to a non-Lombok Java bean so Kotlin projects can also use Proto Util.

Also allow adding and customizing of transformations.

### CODE REVIEW

**Review by DustinLiang — APPROVED** [review]
Nice! Future plans for subclass conversions too?

File: proto-util/src/main/java/games/mythical/proto_util/ProtoUtil.java [hunk 1]
```diff
@@ -40,369 +41,416 @@
...
-  private static Optional<Field> getTargetFieldInDto(final Field srcField, final Class<?> targetClass) {
-    final var name = stripUnderscore(srcField.getName());
-    final var fields = targetClass.getDeclaredFields();
-    for (final var field : fields) {
-      final var anno = field.getAnnotation(ProtoField.class);
-      if (anno != null) {
-        final var annotationValue = anno.value();
-        if (name.equals(annotationValue)) {
-          return Optional.of(field);
+        for (final var transformer : transformers) {
```
Comment (kwang-mythical) [inline proto-util/src/main/java/games/mythical/proto_util/ProtoUtil.java:190]: wonder if there's a clean map way of doing this.

**Review by kwang-mythical — APPROVED** [review]
didnt realize kotlin projects doesn't use lombok.

File: proto-util/src/main/java/games/mythical/proto_util/ProtoUtil.java (hunk 1 above)
Comment (erik-mythical) [reply proto-util/src/main/java/games/mythical/proto_util/ProtoUtil.java:190]: Could do a find first and then return the transform if one is found, but I'm not sure that would be cleaner.

**Pull request merged**